### PR TITLE
[CN-902] change WanReplication validation

### PR DESCRIPTION
**Pull request by SeriyBg**
## User Impact

Added webhook validation for WanReplication update and changed the error messages to be consistent with other validations

### CODE REVIEW

File: controllers/hazelcast/wanreplication_controller.go [hunk 1]
```diff
@@ -120,16 +120,8 @@ func (r *WanReplicationReconciler) Reconcile(ctx context.Context, req ctrl.Reque
...
-		lastSpec := &hazelcastv1alpha1.WanReplicationSpec{}
-		err = json.Unmarshal([]byte(s), lastSpec)
-		if err != nil {
-			err = fmt.Errorf("error unmarshaling Last WanReplication Spec: %w", err)
-			return updateWanStatus(ctx, r.Client, wan, recoptions.Error(err), withWanRepFailedState(err.Error()))
-		}
-
-		err = validateNotUpdatableFields(&wan.Spec, lastSpec)
-		if err != nil {
+	if _, successfullyAppliedBefore := wan.ObjectMeta.Annotations[n.LastSuccessfulSpecAnnotation]; successfullyAppliedBefore {
```
Comment (semihbkgr) [inline controllers/hazelcast/wanreplication_controller.go:123]: Is it really required to check if `LastSuccessfulSpecAnnotation` exists in annotations? `validateNonUpdatableWanFields` function already checks it.

File: api/v1alpha1/wanreolication_validation.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+	if !ok {
+		return allErrs
+	}
+	lastSpec := &WanReplicationSpec{}
+	err := json.Unmarshal([]byte(last), lastSpec)
+	if err != nil {
+		return field.ErrorList{field.InternalError(field.NewPath("spec"), fmt.Errorf("error parsing last WanReplication spec for update errors: %w", err))}
+	}
+
+	if w.Spec.TargetClusterName != lastSpec.TargetClusterName {
```
Comment (semihbkgr) [inline api/v1alpha1/wanreolication_validation.go]: would be better to have separate function which validates unmodifiable fields.

**Review by semihbkgr — APPROVED** [review]
LGTM 👍 with some refactoring.

File: controllers/hazelcast/wanreplication_controller.go (hunk 1 above)
Comment (SeriyBg) [reply controllers/hazelcast/wanreplication_controller.go]: You are right. Removed it

**Pull request merged**